src/components/GoodList/GoodList.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import './GoodList.scss';
+
+type Props = {
+  goods: string[];
+};
+
+type State = {
+  showList: boolean;
+};
+
+export class GoodList extends React.Component<Props, State> {
+  state = {
+    showList: false,
+  };
```

Comment:
remove state showList state and start button from GoodList component and add it to App
so you can conditionally render GoodList
`showList && <GoodList />`